### DISCUSSION TITLE
opening the page with url from server side

### DISCUSSION THREAD

**Discussion by ahmetkuslular** [remix-run/remix#9259] [Q&A]
Hi. I didn't know what to write for the title. In short, here's what I want to do. 

The project is a server side project. When users come to the /users link, the mechanism on the project converts it to /new-users and sends a request to the project in this way. (there is a mechanism like url gatewey) So the req.url comes as /new-users. here the client side needs a routing called /users when rendering. since there is no such route defined, the page cannot be opened. 

To summarise, although /users is written in the browser, the /new-users page defined in the project should be opened. I hope I have explained it correctly. how can I do this?

**Comment by sergiodxa** [comment]
You need to redirect the user, there's a redirect function in Remix you can use to create a redirect response, then return it from a loader.

**Comment by ahmetkuslular** [comment]
I wanted to open the question on the react-router repo, but I opened it here because the wrong link was given. so I'm closing it.

**Reply by sergiodxa** [reply]
The answer is similar, your server should redirect the user to the correct path instead of changing the URL server side

**Reply by ahmetkuslular** [reply]
I think it's the right thing to do, but unfortunately it's a business decision. 🥲

**Reply by kiliman** [reply]
What you're asking for is called URL Rewriting. Unfortunately, React Router and Remix do not support it, as the routes must match the URL that is in the browser address bar.

If your server project is being used as an API server, you could use loaders/clientLoaders to call the `/new-users` API endpoint. This way, the browser still shows `/user,` but the server processes the route it needs.
